From c5680ec1c81fd841d638968988757bd0abfb9d2f Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Tue, 28 Feb 2023 10:59:20 +0000
Subject: [PATCH 1/3] L4TLauncher: boot syslinux instead of extlinux for ostree
 support

Ostree uses syslinux.cfg instead of extlinux.conf for providing the
required kernel/initrd/dtb files for the bootloader.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
Signed-off-by: Jose Quaresma <jose.quaresma@foundries.io>
---
 Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c | 12 +-----------
 Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.h |  6 +++---
 2 files changed, 4 insertions(+), 14 deletions(-)

diff --git a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
index 5d72ea22..5e4ac939 100644
--- a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
+++ b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.c
@@ -1222,11 +1222,6 @@ ProcessExtLinuxConfig (
       if ((BootConfig->NumberOfBootOptions <= MAX_EXTLINUX_OPTIONS) &&
           (BootConfig->NumberOfBootOptions != 0))
       {
-        Status = CheckCommandString (CleanLine, EXTLINUX_KEY_MENU_LABEL, &BootConfig->BootOptions[BootConfig->NumberOfBootOptions-1].MenuLabel);
-        if (!EFI_ERROR (Status)) {
-          continue;
-        }
-
         Status = CheckCommandString (CleanLine, EXTLINUX_KEY_LINUX, &BootConfig->BootOptions[BootConfig->NumberOfBootOptions-1].LinuxPath);
         if (!EFI_ERROR (Status)) {
           continue;
@@ -1347,7 +1342,7 @@ ExtLinuxBootMenu (
   }
 
   for (Index = 0; Index < BootConfig->NumberOfBootOptions; Index++) {
-    Print (L"%d: %s\r\n", Index, BootConfig->BootOptions[Index].MenuLabel);
+    Print (L"%d: %s\r\n", Index, BootConfig->BootOptions[Index].Label);
   }
 
   Status = gBS->SetTimer (EventArray[0], TimerRelative, EFI_TIMER_PERIOD_SECONDS (BootConfig->Timeout)/10);
@@ -2432,11 +2427,6 @@ L4TLauncher (
         FreePool (ExtLinuxConfig.BootOptions[Index].LinuxPath);
         ExtLinuxConfig.BootOptions[Index].LinuxPath = NULL;
       }
-
-      if (ExtLinuxConfig.BootOptions[Index].MenuLabel != NULL) {
-        FreePool (ExtLinuxConfig.BootOptions[Index].MenuLabel);
-        ExtLinuxConfig.BootOptions[Index].MenuLabel = NULL;
-      }
     }
 
     if (ExtLinuxConfig.MenuTitle != NULL) {
diff --git a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.h b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.h
index fc46a70d..241b827b 100644
--- a/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.h
+++ b/Silicon/NVIDIA/Application/L4TLauncher/L4TLauncher.h
@@ -17,7 +17,7 @@
 #define GRUB_BOOTCONFIG_CONTENT_FORMAT  "set cbootargs=\"%s\"\r\nset root_partition_number=%u\r\nset bootimg_present=%u\r\nset recovery_present=%u\r\n"
 #define DETACHED_SIG_FILE_EXTENSION     L".sig"
 
-#define EXTLINUX_CONF_PATH  L"boot\\extlinux\\extlinux.conf"
+#define EXTLINUX_CONF_PATH  L"boot\\loader\\syslinux.cfg"
 
 #define BOOTMODE_DIRECT_STRING    L"bootmode=direct"
 #define BOOTMODE_GRUB_STRING      L"bootmode=grub"
@@ -42,9 +42,9 @@
 #define EXTLINUX_KEY_MENU_TITLE  L"MENU TITLE"
 #define EXTLINUX_KEY_LABEL       L"LABEL"
 #define EXTLINUX_KEY_MENU_LABEL  L"MENU LABEL"
-#define EXTLINUX_KEY_LINUX       L"LINUX"
+#define EXTLINUX_KEY_LINUX       L"KERNEL"
 #define EXTLINUX_KEY_INITRD      L"INITRD"
-#define EXTLINUX_KEY_FDT         L"FDT"
+#define EXTLINUX_KEY_FDT         L"DEVICETREE"
 #define EXTLINUX_KEY_APPEND      L"APPEND"
 #define EXTLINUX_KEY_OVERLAYS    L"OVERLAYS"
 
-- 
2.43.0

